source/PlayerInfo.cpp

```diff
@@ -2621,24 +2621,266 @@ void PlayerInfo::RegisterDerivedConditions()
+			if((!planet && !ship->IsParked()) || (planet && ship->GetPlanet() == planet))
+				retVal += ship->OutfitCount(outfit);
+		}
+		return retVal;
+	});
+
+	// The following condition checks the player's entire fleet for installed outfits.
+	auto &&allInstalledOutfitProvider = conditions.GetProviderPrefixed("outfit (all installed): ");
+	allInstalledOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		const Outfit *outfit = GameData::Outfits().Find(name.substr(strlen("outfit (all installed): ")));
+		if(!outfit)
+			return 0;
+		int64_t retVal = 0;
+		for(const shared_ptr<Ship> &ship : ships)
+			if(!ship->IsDestroyed())
+				retVal += ship->OutfitCount(outfit);
+		return retVal;
+	});
+
+	// The following condition checks the flagship's installed outfits.
+	auto &&flagshipInstalledOutfitProvider = conditions.GetProviderPrefixed("outfit (flagship installed): ");
+	flagshipInstalledOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		if(!flagship)
+			return 0;
+		const Outfit *outfit = GameData::Outfits().Find(name.substr(strlen("outfit (flagship installed): ")));
+		if(!outfit)
+			return 0;
+		return flagship->OutfitCount(outfit);
+	});
+
+	// The following condition checks the player's fleet for outfits in the cargo of escorts
+	// local to the player.
+	auto &&presentCargoOutfitProvider = conditions.GetProviderPrefixed("outfit (cargo): ");
+	presentCargoOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		const Outfit *outfit = GameData::Outfits().Find(name.substr(strlen("outfit (cargo): ")));
+		if(!outfit)
+			return 0;
+		int64_t retVal = 0;
+		if(planet)
+			retVal += Cargo().Get(outfit);
+		for(const shared_ptr<Ship> &ship : ships)
+		{
+			if(ship->IsDestroyed() || ship->GetActualSystem() != system)
+				continue;
+			// If not on a planet, parked ships don't count. If on a planet, the ship's planet must match.
+			if((!planet && !ship->IsParked()) || (planet && ship->GetPlanet() == planet))
+				retVal += ship->Cargo().Get(outfit);
+		}
+		return retVal;
+	});
+
+	// The following condition checks all cargo locations in the player's fleet.
+	auto &&allCargoOutfitProvider = conditions.GetProviderPrefixed("outfit (all cargo): ");
+	allCargoOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		const Outfit *outfit = GameData::Outfits().Find(name.substr(strlen("outfit (all cargo): ")));
+		if(!outfit)
+			return 0;
+		int64_t retVal = 0;
+		if(planet)
+			retVal += Cargo().Get(outfit);
+		for(const shared_ptr<Ship> &ship : ships)
+			if(!ship->IsDestroyed())
+				retVal += ship->Cargo().Get(outfit);
+		return retVal;
+	});
+
+	// The following condition checks the flagship's cargo or the pooled cargo if landed.
+	auto &&flagshipCargoOutfitProvider = conditions.GetProviderPrefixed("outfit (flagship cargo): ");
+	flagshipCargoOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		if(!flagship)
```

Comment:
The flagship variable might not contain the flagship when the player is landed, so the check might just return here and not check the pooled cargo as is specified in the comment.